unit_tests/UnitTestLinearSystem.h

```diff
@@ -262,6 +262,8 @@ class TestLinearSystem : public sierra::nalu::LinearSystem
     return newDeviceCoeffApplier;
   }
 
+  bool owns_coeff_applier() override { return false; }
```

Comment:
So was this missing implementation the source of the leak?

Reply:
Yes. There is code elsewhere that frees the coeff-applier or not, based on whether it is owned by a LinearSystem object. This unit-test one is not owned.